src/components/va-pagination/va-pagination.tsx

```diff
@@ -0,0 +1,236 @@
+  /**
+   * Text added to aria-labels. Any included `{page}` will be replaced with the
+   * associated page number
+   */
+  @Prop() linkAriaLabel: string = 'Load page {page}';
+
+  /**
+   * Clamped props passed to functions
+   * @typedef Options
+   * @property {number} current - current page
+   * @property {number} total - total pages
+   * @property {number} edges - pages around first & last page
+   * @property {number} visible - max number of visible pages
+   */
+  /**
+   * Validate settings & return clamped results
+   */
+  private checkValidation = () => {
+    const total = Math.max(this.total, 1);
+    const current = Math.min(Math.max(this.page, 1), total);
+
+    // Set to a max of 3 to match the default of 9 maxVisible. We probably
+    // wouldn't want to show more than 3 edge pages anyway. It would get really
+    // messy on small screens
+    const edges = Math.min(Math.max(this.edges, 0), 3);
+
+    // edges * 2 + 2 ellipsis + 1 visible active link in the middle
+    const minVisible = Math.min(edges * 2 + 2 + 1, total);
+    const visible = Math.max(this.maxVisible, minVisible);
+
+    return { total, current, edges, visible };
+  }
+
+  /**
+   * Set current page state & emit pageSelect event
+   */
+  private onPageSelected = (page: number) => {
+    this.page = page;
+    this.pageSelect.emit(page);
+  };
+
+  /**
+   * Return common link properties
+   */
+  private getLinkProps = ({ page, current = false, className = '' }) => ({
+    href: '#',
+    class: className,
+    'aria-label': (this.linkAriaLabel || '').replace(/\{page\}/g, page),
+    'aria-current': current ? 'page' : null,
+    onClick: (event: MouseEvent) => {
+      event.preventDefault();
+      this.page = page;
+      this.onPageSelected(page);
+    },
+  });
+
+  
+  /**
+   * Add ellipsis separator
+   */
+  private addEllipsis = () =>
+   <li aria-hidden="true">
+     <span class='ellipsis'>{'\u2026'}</span>
+   </li>;
+
+  /**
+   * Build end (prev or next) link. Renders a span when there is no next page to
+   * navigate to
+   */
+  private buildEndLink = (className: string|null, page: number) => {
+    const isNext = className === 'next';
+    const content = (
+      <Fragment>
+        <span class="vads-u-visibility--screen-reader">
+          {isNext ? 'Next' : 'Previous'}
+        </span>
+        {!isNext && <span aria-hidden="true">{'\u2039\u00a0\u00a0'}</span>}
+        <span class="hide-narrow">{isNext ? 'Next' : 'Prev'}</span>
+        {isNext && <span aria-hidden="true">{'\u00a0\u00a0\u203a'}</span>}
+      </Fragment>
+    );
+    return page ? (
+      <li><a {...this.getLinkProps({ className, page })}>{content}</a></li>
+    ) : (
+      <li aria-hidden="true"><span class={className}>{content}</span></li>
+    );
+  };
+
+  /**
+   * Build page links
+   */
+  private buildLink = (index: number, current: number, total: number) => {
+    let page = index + 1;
+    if (page < 1) {
+      page = 1;
```

Comment:
It seems like this `if` would only happen when `index` is a negative - is that possible?